pkg/querier/querier.go

```diff
@@ -1206,10 +1203,15 @@ func parseDetectedFields(ctx context.Context, limit uint32, streams logqlmodel.S
 					continue
 				}
 
+				if !slices.Contains(df.parsers, *parser) {
+					df.parsers = append(df.parsers, *parser)
+				}
+
 				for _, v := range vals {
 					parsedFields := detectedFields[k]
-					if !parsedFields.isTypeDetected {
+					if detectType {
```

Comment:
this is tricky. just to be clear, out assumption here is that fields in the same stream will have the same type, but fields in different streams may have different types which is why we only include the default "string" type logic in the merge code?

I think that's probably a safe assumption, and it save us from having to run the determine type logic on every line, but I want to make sure it's clear to future readers the assumption we're making.

Reply:
Agree, added a comment!